packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2501,5 +2507,236 @@ describe("Runtime", () => {
 			);
 			assert.equal(mockLogger.events.length, 0, "Expected no more events logged");
 		});
+
+		describe("Signals", () => {
+			let containerRuntime: ContainerRuntime;
+			let logger: MockLogger;
+			let droppedSignals: ISignalEnvelope[];
+
+			// Restore native timer function for signal tests as they rely on Date.now() timestamps
+			before(() => {
+				clock.restore();
+			});
+
+			after(() => {
+				clock = useFakeTimers();
+			});
+
+			beforeEach(async () => {
+				logger = new MockLogger();
+				droppedSignals = [];
+				containerRuntime = await ContainerRuntime.loadRuntime({
+					context: getMockContext({}, logger) as IContainerContext,
+					registryEntries: [],
+					existing: false,
+					requestHandler: undefined,
+					runtimeOptions: {
+						enableGroupedBatching: false,
+					},
+					provideEntryPoint: mockProvideEntryPoint,
+				});
+			});
+
+			function sendSignals(numberOfSignals: number) {
+				for (let i = 0; i < numberOfSignals; i++) {
+					containerRuntime.submitSignal("TestSignalType", `TestSignalContent ${i}`);
+				}
+			}
+
+			function processSignals(signals: ISignalEnvelope[], numberOfSignals: number) {
+				const signalsToProcess = signals.splice(0, numberOfSignals);
+				for (const signal of signalsToProcess) {
+					containerRuntime.processSignal(
+						{
+							clientId: containerRuntime.clientId as string,
+							content: signal,
+						},
+						true,
+					);
+				}
+			}
+
+			function dropSignals(numberOfSignals: number) {
+				const signalsToDrop = submittedSignals.splice(0, numberOfSignals);
+				droppedSignals.push(...signalsToDrop);
+			}
```

Comment:
These could use comments.
Since processSignals was updated to be given an array, can we do the same for all?
Perhaps `sendSignals` should return the accumulated signals and all others take signal array.

The maintenance of what array has what can be confusing to follow. Implicit modifications are side-effects that are hard to make sense of.

The alternative is to create a helper class that wraps it all up (and doesn't expose the inner arrays).

Reply:
I went with the helper function to wrap up the arrays and avoid them passing around.